pkg/asset/installconfig/openstack/validation/cloudinfo.go

```diff
@@ -297,3 +319,114 @@ func (ci *CloudInfo) getZones() ([]string, error) {
 
 	return zones, nil
 }
+
+// loadLimits loads the consumer quota metric.
+func loadLimits(opts *clientconfig.ClientOpts) ([]record, error) {
+	var limits []record
+
+	projectID, err := getProjectID(opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get keystone project ID")
+	}
+
+	computeRecords, err := getComputeLimits(opts, projectID)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get compute quota records")
+	}
+	for _, r := range computeRecords {
+		limits = append(limits, r)
+	}
+
+	return limits, nil
+}
+
+func getComputeLimits(opts *clientconfig.ClientOpts, projectID string) ([]record, error) {
+	computeClient, err := clientconfig.NewServiceClient("compute", opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to connect against OpenStack Comute v2 API")
+	}
+	qs, err := computequotasets.GetDetail(computeClient, projectID).Extract()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get QuotaSets from OpenStack Compute API")
+	}
+
+	var records []record
+	addRecord := func(name string, quota computequotasets.QuotaDetail) {
+		records = append(records, record{
+			Service: "compute",
+			Name:    name,
+			Value:   int64(quota.Limit - quota.InUse - quota.Reserved),
+		})
+	}
+	addRecord("Cores", qs.Cores)
+	addRecord("Instances", qs.Instances)
+	addRecord("RAM", qs.RAM)
+
+	return records, nil
+}
+
+func getProjectID(opts *clientconfig.ClientOpts) (string, error) {
+	keystoneClient, err := clientconfig.NewServiceClient("identity", opts)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to conect against OpenStack Keystone API")
+	}
+	authResult := keystoneClient.GetAuthResult()
+	if authResult == nil {
+		return "", errors.Errorf("Client did not use openstack.Authenticate()")
+	}
+
+	switch authResult.(type) {
+	case tokensv2.CreateResult:
+		// Gophercloud has support for v2, but keystone has deprecated
+		// and it's not even documented.
+		return "", errors.Errorf("Extracting project ID using the keystone v2 API is not supported")
+
+	case tokensv3.CreateResult:
+		v3Result := authResult.(tokensv3.CreateResult)
+		project, err := v3Result.ExtractProject()
+		if err != nil {
+			return "", errors.Wrap(err, "Extracting project from v3 authResult")
+		} else if project == nil {
+			return "", errors.Errorf("Token is not scoped to a project")
+		}
+		return project.ID, nil
+
+	default:
+		return "", errors.Errorf("Unsupported AuthResult type: %T", authResult)
+	}
+}
+
+// LoadQuotas loads the quota information for a project and provided services. It provides information
+// about the usage and limit for each resource quota..
+func LoadQuotas(opts *clientconfig.ClientOpts) ([]quota.Quota, error) {
```

Comment:
This function doesn't have to be public.
```suggestion
func loadQuotas(opts *clientconfig.ClientOpts) ([]quota.Quota, error) {
```